public/components/datasources/components/manage/accelerations/acceleration_details_flyout.tsx

```diff
@@ -24,20 +24,90 @@ import {
   onRefreshButtonClick,
   onDiscoverButtonClick,
   onDeleteButtonClick,
-} from '../accelerations/helpers/utils';
+} from './utils/acceleration_utils';
+import { coreRefs } from '../../../../../framework/core_refs';
+import { OpenSearchDashboardsResponse } from '../../../../../../../../src/core/server/http/router';
 
 export interface AccelerationDetailsFlyoutProps {
   acceleration: any;
 }
 
-export const AccelerationDetailsFlyout = (props: AccelerationDetailsFlyoutProps) => {
-  const { acceleration } = props;
+const getMappings = (index: string): Promise<OpenSearchDashboardsResponse> | undefined => {
+  console.log('getMappings index: ', index);
+  return coreRefs.dslService?.fetchFields(index);
+};
+
+const getSettings = (index: string): Promise<OpenSearchDashboardsResponse> | undefined => {
+  return coreRefs.dslService?.fetchSettings(index);
+};
+
+const getIndexInfo = (index: string): Promise<OpenSearchDashboardsResponse> | undefined => {
+  return coreRefs.dslService?.fetchIndices(index);
+};
+
+const handlePromise = (
```

Comment:
feel like we should use a more descriptive name for this function

Reply:
changed to `handleDetailsFetchingPromise`